tools/cmake/UploadMethods.md

```diff
@@ -0,0 +1,214 @@
+Mbed OS's CMake build system supports a number of upload methods to help get code onto your device.  Some uploaders also support debugging, and for these CMake will help you by automatically running the needed commands to start a debugging session.
+
+In order to enable uploading, you'll need to configure which upload methods your project supports.  To do this, you'll need to create an UploadMethodConfig.cmake file and include it in your top-level CMakeLists.txt.  You can start by using the template UploadMethodConfig-Sample.cmake (which is for Nucleo F429ZI) in this folder.  You'll then edit it to control the upload methods that are available for your target, and set the parameters that they need to operate.  THe different parameters and their values are explained in this document.
+
+This page lists each upload method, its _parameters_ (variables that will have a fixed value) and its _options_ (variables that are different for different people and should be set on the command line when you run cmake).  Since this config is set by standard CMake code, it's very easy to add your own additional logic (including command line options) to adjust these variables!
```

Comment:
double space
```suggestion
This page lists each upload method, its _parameters_ (variables that will have a fixed value) and its _options_ (variables that are different for different people and should be set on the command line when you run cmake). Since this config is set by standard CMake code, it's very easy to add your own additional logic (including command line options) to adjust these variables!
```